src/components/Dialog/Dialog.ts

```diff
@@ -46,99 +55,85 @@ export type NotifyAndNotShowAgainButtonOption = {
   options.cancel ??= "キャンセル";
 
-  return new Promise((resolve: CommonDialogCallback) => {
-    setCommonDialogCallback(
-      Dialog.create({
-        title: options.title,
-        message: options.message,
-        persistent: true, // ダイアログ外側押下時・Esc押下時にユーザが設定ができたと思い込むことを防止する
-        focus: "ok",
-        html: options.html,
-        ok: {
-          flat: true,
-          label: options.actionName,
-          textColor: "display",
-        },
-        cancel: {
-          flat: true,
-          label: options.cancel,
-          textColor: "display",
-        },
-      }),
-      resolve,
-    );
-  });
+  const { promise, resolve } = Promise.withResolvers<number>();
+  Dialog.create({
+    component: QuestionDialog,
+    componentProps: {
+      type: options.type ?? "question",
+      title: options.title,
+      message: options.message,
+      buttons: [options.cancel, options.actionName],
+      default: 0,
+    },
+  }).onOk(({ index }: { index: number }) => resolve(index));
+
+  const index = await promise;
+
+  return index === 1 ? "OK" : "CANCEL";
 };
 
-export const showWarningDialog = async (
-  options: CommonDialogOptions["warning"],
-) => {
+export const showWarningDialog = async (options: TextWarningDialogOptions) => {
   options.cancel ??= "キャンセル";
 
-  return new Promise((resolve: CommonDialogCallback) => {
-    setCommonDialogCallback(
-      Dialog.create({
-        title: options.title,
-        message: options.message,
-        persistent: true,
-        focus: "cancel",
-        ok: {
-          label: options.actionName,
-          flat: true,
-          textColor: "warning",
-        },
-        cancel: {
-          label: options.cancel,
-          flat: true,
-          textColor: "display",
-        },
-      }),
-      resolve,
-    );
-  });
+  const { promise, resolve } = Promise.withResolvers<number>();
+  Dialog.create({
+    component: QuestionDialog,
+    componentProps: {
+      type: options.type ?? "warning",
+      title: options.title,
+      message: options.message,
+      buttons: [options.cancel, options.actionName],
+      default: 0,
+    },
+  }).onOk(({ index }: { index: number }) => resolve(index));
+
+  const index = await promise;
+
+  return index === 1 ? "OK" : "CANCEL";
 };
-
-const setCommonDialogCallback = (
-  dialog: DialogChainObject,
-  resolve: (result: CommonDialogResult) => void,
+export const showQuestionDialog = async (
+  options: TextQuestionDialogOptions,
 ) => {
-  return dialog
-    .onOk(() => {
-      resolve("OK");
-    })
-    .onCancel(() => {
-      resolve("CANCEL");
-    });
+  const { promise, resolve } = Promise.withResolvers<number>();
+  Dialog.create({
+    component: QuestionDialog,
+    componentProps: {
+      type: options.type ?? "question",
+      title: options.title,
+      message: options.message,
+      buttons: options.buttons,
+      persistent: options.cancel != undefined,
```

Comment:
ここ逆かも･･･？
（trueだとESCとかで閉じられなくなる･･･？）
```suggestion
      persistent: options.cancel == undefined,
```